dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/AbstractRegistry.java

```diff
@@ -228,25 +238,21 @@ private void loadProperties() {
     public List<URL> lookup(URL url) {
         List<URL> result = new ArrayList<>();
         Map<String, List<URL>> notifiedUrls = getNotified().get(url);
-        if (notifiedUrls != null && notifiedUrls.size() > 0) {
+        if (CollectionUtils.isNotEmptyMap(notifiedUrls)) {
             for (List<URL> urls : notifiedUrls.values()) {
-                for (URL u : urls) {
-                    if (!Constants.EMPTY_PROTOCOL.equals(u.getProtocol())) {
-                        result.add(u);
-                    }
-                }
+                urls.stream().filter(u -> !Constants.EMPTY_PROTOCOL.equals(u.getProtocol())).forEach(u -> {
+                    result.add(u);
+                });
             }
         } else {
             final AtomicReference<List<URL>> reference = new AtomicReference<>();
             NotifyListener listener = reference::set;
             subscribe(url, listener); // Subscribe logic guarantees the first notify to return
             List<URL> urls = reference.get();
             if (CollectionUtils.isNotEmpty(urls)) {
-                for (URL u : urls) {
-                    if (!Constants.EMPTY_PROTOCOL.equals(u.getProtocol())) {
-                        result.add(u);
-                    }
-                }
+                urls.stream().filter(u -> !Constants.EMPTY_PROTOCOL.equals(u.getProtocol())).forEach(u -> {

```

Comment:
What do you think about one stream method per line?
```suggestion
                urls.stream().filter(u -> !Constants.EMPTY_PROTOCOL.equals(u.getProtocol()))
                .forEach(u -> {
```

Reply:
Emmm, your suggestion looks good, thank you for your suggestion.